app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,52 @@
+class ArticlesController < ApplicationController
+  
+  http_basic_authenticate_with name: "mayank", password: "bv-mayank", except: [:index, :show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article = Article.find(params[:id])
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+
+    if @article.save
+      redirect_to @article
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
```

Comment:
```suggestion
  def create
    @article = Article.new(article_params)
    @article.save! ? redirect_to @article : render(:new, status: :unprocessable_entity)
  rescue => e
    logger.error "CREATE_NEW_ARTICLE_ERROR: #{e.message} : #{e.class} : #{e.backtrace}"
    render :new, status: :unprocessable_entity
  end
```
- It is better to use save with `!`, read about it.
- handle error in case of failure and log error in case of error